mutation-testing/output-pr-mutants/action.yml

```
@@ -0,0 +1,166 @@
+name: Output Mutants
+
+runs:
+  using: "composite"
+
+  steps:
+    - name: Download artifacts
+      uses: actions/download-artifact@v3
+
+    - name: Append output from shards
+      shell: bash
+      run: |
+        # List of all possible mutant outcomes file names and folder names collected from jobs.
+        folders=("mutants-shard-big--1" "mutants-shard-big-0" "mutants-shard-big-1" "mutants-shard-big-2" "mutants-shard-big-3" "mutants-shard-big-4" "mutants-shard-big-5" "mutants-shard-big-6" "mutants-shard-big-7" "mutants-shard-small--1" "mutants-shard-small-0" "mutants-shard-small-1" "mutants-shard-small-2" "mutants-shard-small-3")
+        files=("missed.txt" "caught.txt" "timeout.txt" "unviable.txt")
+        mkdir -p mutants-shards
+
+        # If the folder/file path exists, append the output to it's corresponding file name in a newly created folder that will contain all outputs combined
+        for file in "${files[@]}"; do
+          for folder in "${folders[@]}"; do
+            if [[ -s "$folder/$file" ]]; then
+              cat "$folder/$file" >> "mutants-shards/$file"
+            fi
```

```suggestion
            if [[ -s "$folder/$file" ]]; then
              cat "$folder/$file" >> "mutants-shards/$file"
            else
              echo "Missing file ($folder/$file)" 
            fi
```
not sure if this step should exit if a file is not found though

---

If i'm running without shards for each folder from 0 to 7 (big packages), respectively for 0 to 3 (small) i will get the 'Missing file' print 4 times for 12 folders. Only the artifacts with '-1' will be available and have outputs, so the print from the 'else' branch would be irrelevant. Also, if i'm running with shards, it will say 'Missing file' 4 times for each '-1' folder.

---

> not sure if this step should exit if a file is not found though

There will always be at least 2 missing folders due to the fact that it selects shards or no-shards from the complete list which includes both of them.

---

Just had an idea, i think we can parse from the main workflow the current case running through inputs ( if there are big packages, with/without shards, same for small ). Then, we know exactly which folders will be uploaded to artifacts and can make a dynamic list containing only the folders we need. That way we can also exit in the else branch since now we only check for folders we are certain should contain these mutant-outcomes files, and by not having them there we are sure something is wrong with the flow.

Referenced to this: https://github.com/stacks-network/actions/pull/8#discussion_r1449602978